core/tests/protractor_utils/TopicViewerPage.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the topic and story viewer page, for use
+ * in Protractor tests.
+ */
+
+var waitFor = require('./waitFor.js');
+var action = require('../protractor_utils/action.js');
+
+var TopicViewerPage = function() {
+  var topicDescription = element(by.css('.protractor-test-topic-description'));
+  var storySummaryTitleList =
+    element.all(by.css('.protractor-test-story-summary-title'));
+  var topicLinkList = element.all(by.css('.protractor-test-topic-link'));
+
+  this.get = async function(classroomUrlFragment, topicIndex) {
+    await browser.get(`/learn/${classroomUrlFragment}`);
+    await waitFor.pageToFullyLoad();
+    var topicLink = topicLinkList.get(topicIndex);
+    await action.click('Topic TASV1', topicLink);
```

Comment:
Can you identify topics by their names instead of index? This makes the tests a lot more readable.

`TASV1` is specific to a particular topic, but this utility might be used in other contexts. Can you use the true name? You'll have access to this once you identify the topic by name instead of index.

Reply:
What do you mean by true name?

Currently I have done this, 
```
await topicViewerPage.get('math', 'Topic TASV1');
```
&
```
  this.get = async function(classroomUrlFragment, topicName) {
    await browser.get(`/learn/${classroomUrlFragment}`);
    await waitFor.pageToFullyLoad();
    var topicLink = element(by.cssContainingText(
      '.protractor-test-topic-link', new RegExp(topicName)));
    if (await topicLink.isPresent()) {
      await action.click('Topic TASV1', topicLink);
      await waitFor.pageToFullyLoad();
    }
  };
```

Would this work?

Reply:
Sorry, that was a typo. I meant "topic name". Your code looks reasonable to me, though I think we should not silently ignore cases where the topic is missing. These indicate a problem and should raise an error. Also, use the actual topic name instead of `Topic TASV1`.

Reply:
Done.